tests/kernel/nfs_server.pm

```diff
@@ -0,0 +1,101 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: NFS server
+#    This module provisions the NFS server and then runs some
+#    basic sanity tests
+# Maintainer: Kernel QE <kernel-qa@suse.de>
+
+use Mojo::Base "opensusebasetest";
+use testapi;
+use serial_terminal "select_serial_terminal";
+use lockapi;
+use utils;
+use Utils::Logging "export_logs_basic";
+
+sub run {
+    select_serial_terminal();
+
+    #TODO: configure nfs config as pleased, specifically:
+    #USE_KERNEL_NFSD_NUMBER
+    #NFS3_SERVER_SUPPORT
+    #NFS4_SUPPORT
+    #NFSV4LEASETIME
+    #TODO: introduce nfs ha setup
+
+    #TODO: all below folder creation and exports should be abstracted
+    #TODO: allow this to be configurable
+    my $nfs_mount_nfs3 = "/nfs/shared_nfs3";
+    my $nfs_mount_nfs4 = "/nfs/shared_nfs4";
+    my $nfs_permissions = "rw,sync,no_root_squash";
+
+    my $nfs_mount_nfs3_async = "/nfs/shared_nfs3_async";
+    my $nfs_mount_nfs4_async = "/nfs/shared_nfs4_async";
+    my $nfs_permissions_async = "rw,async,no_root_squash";
+
+    # provision NFS server(s) of various types
+    zypper_call("in yast2-nfs-server");
+
+    assert_script_run("mkdir -p $nfs_mount_nfs3");
+    assert_script_run("chmod 777 $nfs_mount_nfs3");
+    assert_script_run("mkdir -p $nfs_mount_nfs4");
+    assert_script_run("chmod 777 $nfs_mount_nfs4");
+
+    assert_script_run("mkdir -p $nfs_mount_nfs3_async");
+    assert_script_run("chmod 777 $nfs_mount_nfs3_async");
+    assert_script_run("mkdir -p $nfs_mount_nfs4_async");
+    assert_script_run("chmod 777 $nfs_mount_nfs4_async");
+
+    assert_script_run("echo $nfs_mount_nfs3 client-node00\\($nfs_permissions\\) >> /etc/exports");
+    assert_script_run("echo $nfs_mount_nfs4 client-node00\\($nfs_permissions\\) >> /etc/exports");
+    assert_script_run("echo $nfs_mount_nfs3_async client-node00\\($nfs_permissions_async\\) >> /etc/exports");
+    assert_script_run("echo $nfs_mount_nfs4_async client-node00\\($nfs_permissions_async\\) >> /etc/exports");
+
+    record_info("EXPORTS", script_output("cat /etc/exports"));
+
+    systemctl("enable nfs-server");
```

Comment:
Also restart rpcbind before starting nfs-server, maybe safer.

Reply:
I needed to restart rpcbind to get LTP tests running.

Reply:
yeah it might be but I kept it this way to see how this really behaves. If you don't mind I keep it this way atm but sure this is something we might be doing eventually (restart of the rpc)